packages/components/CONTRIBUTING.md

```diff
@@ -236,6 +232,95 @@ A couple of good examples of how hooks are used for composition are:
 
 TDB -->
 
+## Naming Conventions
+
+It is recommended that compound components use dot notation to separate the namespace from the individual component names. The top-level compound component should be called the namespace (no dot notation).
+
+Dedicated React context should also use dot notation, while hooks should not.
+
+When exporting compound components and preparing them to be consumed, it is important that:
+
+-   the JSDocs appear correctly in IntelliSense;
+-   the top-level component's JSDoc appears in the Storybook docs page;
+-   the top-level and subcomponent's prop types appear correctly in the Storybook props table.
+
+To meet the above requirements, we recommend:
+
+-   using `Object.assign()` to add subcomponents as properties of the top-level component;
+-   using named functions for all components;
+-   setting explicitly the `displayName` on all subcomponents;
+-   adding the top-level JSDoc to the result of the `Object.assign` call;
+-   adding inline subcomponent JSDocs inside the `Object.assign` call.
+
+The following example implements all of the above recommendations.
+
+```tsx
+//=======================
+// Component.tsx
+//=======================
+import { forwardRef, createContext } from '@wordpress/element';
+
+const UnforwardedTopLevelComponent = function TopLevel( props, ref ) {
+	/* ... */
+};
+const TopLevelComponent = forwardRef( UnforwardedTopLevelComponent );
+
+const UnforwardedSubComponent = function Sub( props, ref ) {
```

Comment:
I don't see a good reason to go with `const X = function Y()` syntax, when we can just do `function X`. 

Seems like we can update that before merging.

```suggestion
function UnforwardedTopLevelComponent( props, ref ) {
	/* ... */
};
const TopLevelComponent = forwardRef( UnforwardedTopLevelComponent );

function UnforwardedSubComponent( props, ref ) {
```